src/test/scala/com/fulcrumgenomics/bam/ErrorRateByReadPositionTest.scala

```diff
@@ -25,229 +25,170 @@
-    metrics.flatten.foreach { metric =>
-      if (metric.position <= 10) {
-        metric shouldBe new ErrorRateByReadPositionMetric(0, metric.position, 0.0d, 1l)
-      }
-      else {
-        metric shouldBe new ErrorRateByReadPositionMetric(0, metric.position, 0.0d, 2l)
-      }
+  it should "not count as errors either Ns in reads or ambiguous bases in the reference" in {
+    val builder = newSamBuilder
+    builder.addPair(contig=0, start1=100, start2=100).foreach(_.setReadString("A"*20))
+    builder.addPair(contig=1, start1=100, start2=200).foreach(_.setReadString("AANA"*5))
+    val (out, pre) = outputAndPrefix
+    val metrics = new ErrorRateByReadPosition(input=builder.toTempFile(), output=Some(pre), ref=ref).computeMetrics
+    metrics.size shouldBe 40
+    metrics.map(_.bases_total).sum shouldBe 30 // nothing on contig 0, and 3/4 on contig 1
+    metrics.forall(_.error_rate == 0.0) shouldBe true
+  }
+
+  it should "restrict to a set of intervals if provided" in {
+    val builder = newSamBuilder
+    builder.addPair(contig=1, start1=100, start2=150).foreach(_.setReadString("AAAA"*5))
+    builder.addPair(contig=1, start1=200, start2=250).foreach(_.setReadString("AAAA"*5))
+    builder.addPair(contig=1, start1=300, start2=350).foreach(_.setReadString("ACGA"*5))
+    builder.addPair(contig=1, start1=400, start2=450).foreach(_.setReadString("AAAA"*5))
+
+    val intervals = new IntervalList(dict)
+    intervals.add(new Interval("chr1", 100, 275))
+    intervals.add(new Interval("chr1", 400, 500))
+    val intervalPath = makeTempFile("regions.", ".interval_list")
+    intervals.write(intervalPath.toFile)
+
+    val (out, pre) = outputAndPrefix
+    val metrics = new ErrorRateByReadPosition(input=builder.toTempFile(), output=Some(pre), ref=ref, intervals=Some(intervalPath)).computeMetrics
+    metrics.forall(_.error_rate == 0.0) shouldBe true
+    metrics.map(_.bases_total).sum shouldBe (3 * 2 * 20)
+  }
+
+  it should "not count errors that occur at sites with variants" in {
+    val builder = newSamBuilder
+    builder.addFrag(contig=1, start=490).foreach(_.setReadString("GAAAAAAAAAGAAAAAAAAA"))
+    builder.addFrag(contig=2, start=490).foreach(_.setReadString("TCCCCCCCCCTCCCCCCCCC"))
+    builder.addFrag(contig=3, start=490).foreach(_.setReadString("AGGGGGGGGGAGGGGGGGGG"))
+    builder.addFrag(contig=4, start=490).foreach(_.setReadString("CTTTTTTTTTTTTTTTTTTT"))
+
+    val (out, pre) = outputAndPrefix
+    val metrics = new ErrorRateByReadPosition(input=builder.toTempFile(), output=Some(pre), ref=ref, variants=Some(vcf)).computeMetrics
+    metrics.find(_.position == 1).get.error_rate shouldBe 1.0
+    metrics.find(_.position == 11).get.error_rate shouldBe 0.0
+    metrics.filter(m => m.position != 1 && m.position != 11).foreach { m =>
+      m.bases_total shouldBe 4
+      m.error_rate  shouldBe 0.0
     }
   }
 
-  it should "ignore soft-clipped bases at the end of the read" in {
```

Comment:
I assume these are gone since we are assuming SamLocusIterator does this for us?